Modules/_pickle.c

```diff
@@ -7969,8 +7969,9 @@ pickle_clear(PyObject *m)
 }
 
 static void
-pickle_free(PyObject *m)
+pickle_free(void *arg)
 {
+    PyObject *m = (PyObject*)arg;
     _Pickle_ClearState(_Pickle_GetState(m));
```

Comment:
```C
pickle_free(void *mod)
{
    PyObject *m = (PyObject*)mod;
    _Pickle_ClearState(_Pickle_GetState(m));
```

I think most of the free functions use m, mod or module for the free function of module. Maybe use `m` and do an inline cast. Or just do:

```c
pickle_free(void *m) {
    (void)pickle_clear((PyObject *)m);
}
```

which is what we usuaslly do.